packages/languages/src/node/language-server-contribution.ts

```diff
@@ -26,14 +28,36 @@ export {
 export const LanguageServerContribution = Symbol('LanguageServerContribution');
 export interface LanguageServerContribution extends LanguageContribution {
     start(clientConnection: IConnection): void;
+    /**
+     * Called on shut down. Only synchronous operations are permitted.
+     */
+    stop?(): void;
 }
 
 @injectable()
 export abstract class BaseLanguageServerContribution implements LanguageServerContribution {
+
     abstract readonly id: string;
     abstract readonly name: string;
+    protected readonly disposables: Disposable[] = [];
+
+    @inject(RawProcessFactory)
+    protected readonly processFactory: RawProcessFactory;
+
+    @inject(ProcessManager)
+    protected readonly processManager: ProcessManager;
+
     abstract start(clientConnection: IConnection): void;
 
+    stop(): void {
+        while (this.disposables.length > 0) {
```

Comment:
if `disposables` is `DisposableCollection`, then you can just call `dispose` here